api/handlers.go

```diff
@@ -89,12 +97,30 @@ func (api *API) unpinPOST(w http.ResponseWriter, req *http.Request, _ httprouter
 }
 
 // sweepPOST instructs pinner to scan the list of skylinks pinned by skyd and
-// update its database.
+// update its database. This call is non-blocking, i.e. it will immediately
+// return with a success and it will only start a new sweep if there isn't one
+// already running. The response is 202 Accepted and the response body contains
+// an endpoint link on which the caller can check the status of the sweep.
 func (api *API) sweepPOST(w http.ResponseWriter, _ *http.Request, _ httprouter.Params) {
-	// TODO This should update both skylinks which are pinned but are not marked
-	//  as pinned by the local server and skylinks which are marked as pinned
-	//  but no longer are.
-	api.WriteError(w, errors.New("unimplemented"), http.StatusTeapot)
+	api.latestSweepStatusMu.Lock()
+	// If there is no sweep in progress - kick one off.
+	if !api.latestSweepStatus.InProgress {
+		go api.threadedPerformSweep()
+	}
+	api.latestSweepStatusMu.Unlock()
+	// TODO If we want to be able to uniquely identify sweeps we can issue ids
+	//  for them and keep their statuses in a map. This would be the appropriate
+	//  RESTful approach. I am not sure we need that because all we care about
+	//  is to be able to kick off one and wait for it to end and this
+	//  implementations is sufficient for that.
```

Comment:
I'm looking for opinions here, that's why the TODO is left in the code.

Reply:
Implementing this would require us to store those runs in the DB, so they would accessible from any server in the cluster.

Actually, we need to do that anyway for this monitoring to be able happen reliably because with the current approach the caller will get unreliable data if they get directed to another server by the LB. So, this definitely needs to live in the DB.

This should be a F/U, though. I would basically remove the endpoint until then or mark it as "only call this locally".